aws/rust-runtime/aws-sigv4/src/http_request/canonical_request.rs

```diff
@@ -774,14 +774,16 @@ mod tests {
         assert_eq!(creq.values.signed_headers().as_str(), "host;x-amz-date");
     }
 
-    // It should exclude user-agent and x-amz-user-agent headers from presigning
+    // It should exclude authorization, user-agent, x-amz-user-agent, x-amzn-trace-id headers from presigning
```

Comment:
Huh, I'm surprised this is only tested for presigning. Would you be willing to copy/paste this test and modify it for non-presigning? The `SignatureLocation` would get changed to `Headers`, `expires_in` would not be set, and `canonical.values.into_query_params()` would change to `canonical.values.as_headers()`. Otherwise, it should be identical.

Reply:
There are a couple other changes here. 

1. x-amz-user-agent actually only gets filtered for presigning: https://github.com/awslabs/smithy-rs/blob/main/aws/rust-runtime/aws-sigv4/src/http_request/canonical_request.rs#L253 so `x-amz-user-agent` is actually still present
2. x-amz-date is added on non-presigning requests